### PR TITLE
MCR-4266 Add setting assignments banner and adjust display name

### PR DESCRIPTION
**Pull request by haworku**
## Summary
CMS or Admin can save analysts added and removed from state

- Follow on to #2779 -  all AC should be covered now. 
- Fixed filtering analysts. Caused crash when selecting the dropdown
- Fixed filtering analysts by `No assignments` see [MCR-4488](https://jiraent.cms.gov/browse/MCR-4488)
- Added state assignments filter tests

#### Related issues
https://jiraent.cms.gov/browse/MCR-4266

#### Screenshots
Screenshot 2024-09-25 at 4 11 46 PM

#### Test cases covered
Did not add additional tests, would like to test the banner in cypress

## QA guidance
Go through the AC on the linked ticket and Figma designs.

### CODE REVIEW

**Review by JasonLin0991 — COMMENTED** [review]
The banner does not show up on the review app, see the comment below, but looks good testing on local. I can review again once you get that banner to show up.

File: services/app-web/src/pages/Settings/EditStateAssign/EditStateAssign.tsx [hunk 1]
```diff
@@ -81,23 +85,28 @@ export const EditStateAssign = (): React.ReactElement => {
...
         )
-
         if (result instanceof Error) {
-            console.error(result)
-            // react should be displaying the error using the updateError var
+            recordJSException(result)
+            // editError will ensure banner is displayed, no need to handle here.
         } else {
-            navigate(RoutesRecord.STATE_ASSIGNMENTS)
+            navigate(`${RoutesRecord.STATE_ASSIGNMENTS}/?submit=state-assignments`)
```
[the remark this answers is not in the document]
Comment (haworku) [reply services/app-web/src/pages/Settings/EditStateAssign/EditStateAssign.tsx:109]: ^ oh okay sure i can do that

File: services/app-web/src/pages/Settings/Settings.tsx [hunk 2]
```diff
@@ -102,10 +116,16 @@ export const Settings = (): React.ReactElement => {
...
         data: mcrSettingsData,
         error: mcReviewError,
+        refetch: refetchMcReviewSettings
     } = useFetchMcReviewSettingsQuery({
         skip: !readWriteStateAssignments,
+        notifyOnNetworkStatusChange: true,
     })
 
+    useEffect(()=> {
+       void refetchMcReviewSettings()
```
[the remark this answers is not in the document]
Comment (haworku) [reply services/app-web/src/pages/Settings/Settings.tsx:126]: Not exactly. We could handle just that with cacheing. It's addressing a bigger issue - settings never refetches once you land on it. The way the API is set up in that outlet (which outlets are usually used for layouts not for APIs) it means that page's data never reloads. We only ever fetch once, even though users could be going to specific pages and coming back several times over. Alsop other users could also be changing settings in background 

At least in the background, the main settings should be re-fetching I thought.  I did something a little more excessive here (fetching every route change) but could change that to just be when the route is on main settings

[the remark this answers is not in the document]
Comment (JasonLin0991) [reply services/app-web/src/pages/Settings/Settings.tsx:126]: This makes sense. The only other thing I can think of is polling.

File: services/app-web/src/pages/Settings/EditStateAssign/EditStateAssign.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (haworku) [reply services/app-web/src/pages/Settings/EditStateAssign/EditStateAssign.tsx:109]: I started to address this then took it off, I wanted to see what happened with the cacheing in place - this is redeploying now

File: services/app-web/src/pages/Settings/Settings.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (haworku) [reply services/app-web/src/pages/Settings/Settings.tsx]: I adjusted it to be even more narrowly called - just calling it if there was a recent update to something.

**Review by JasonLin0991 — APPROVED** [review]

**Pull request merged**